pkg/sources/github/github.go

```diff
@@ -454,6 +416,58 @@ RepoLoop:
 	return nil
 }
 
+// ensureRepoInfoCache checks that s.repoInfoCache has an entry for the
+// provided repository URL. If not, it fetches and stores the metadata for the
+// repository. In some cases, the gist URL needs to be normalized, which is
+// returned by this function.
+func (s *Source) ensureRepoInfoCache(ctx context.Context, repo string) (string, error) {
+	if _, ok := s.repoInfoCache.get(repo); ok {
+		return repo, nil
+	}
+	// Ensure that |s.repoInfoCache| contains an entry for |repo|.
+	// This compensates for differences in enumeration logic between `--org` and `--repo`.
+	// See: https://github.com/trufflesecurity/trufflehog/pull/2379#discussion_r1487454788
```

Comment:
Am I understanding correctly that with this function extracted, this comment is somewhat out of date? It's not exactly _wrong_, but it doesn't tell the whole story anymore, and the information it contains seems like it belongs in the doc comment.